### ISSUE TITLE
Add sam2

### ISSUE THREAD

**Issue by innat** [keras-team/keras-hub#2224]
https://github.com/facebookresearch/sam2

**Comment by mattdangerw** [comment]
@divyashreepathihalli would probably know re inference. If we don't support it we should add it.

I don't see why we wouldn't want SAM 2, so SGTM to add. Are you looking to contribute this one?

**Comment by innat** [comment]
> Are you looking to contribute this one?

Not at the moment. I was exploring options for fine-tuning SAM on 3D data and came across this method, which supports both images and videos.

**Comment by pctablet505** [comment]
@mattdangerw @divyashreepathihalli 
If nobody is working on it, I would like to add this one.

**Comment by tirthasheshpatel** [comment]
> or current implementation only for inference?

Only for inference when I wrote it. I never tested training so I wonder how much effort is required to support it. I can try to train and compare the gradients to PyTorch impl when I get some time.

**Comment by innat** [comment]
@tirthasheshpatel 
Thanks. It would be really helpful.
